```
************* Module payments.payments.mangopay_client
payments/payments/mangopay_client.py:26:31: W9904: gettext-ed string with positional args 'Basic {}' (bad-gettext-ed-string)
payments/payments/mangopay_client.py:60:34: W9904: gettext-ed string with positional args '\nURL: {}\nPayload: {}\nError: {}' (bad-gettext-ed-string)
payments/payments/mangopay_client.py:81:25: W9904: gettext-ed string with positional args 'Transaction report for wallet {}' (bad-gettext-ed-string)
************* Module payments.payments.utils
payments/payments/utils.py:150:18: W9904: gettext-ed string with positional args 'Insurance for {}' (bad-gettext-ed-string)
payments/payments/utils.py:162:14: W9904: gettext-ed string with positional args 'Could not transfer insurance for Payment Request {}. MangoPay error: {}' (bad-gettext-ed-string)
payments/payments/utils.py:198:18: W9904: gettext-ed string with positional args 'Customer service fee for {}' (bad-gettext-ed-string)
payments/payments/utils.py:210:14: W9904: gettext-ed string with positional args 'Could not transfer customer fee for Payment Request {}. MangoPay error: {}' (bad-gettext-ed-string)
payments/payments/utils.py:246:14: W9904: gettext-ed string with positional args "Can't make transfer, recipient has no wallet.Transaction: {} " (bad-gettext-ed-string)
payments/payments/utils.py:288:14: W9904: gettext-ed string with positional args 'Could not transfer Payment Request {}. MangoPay error: {}' (bad-gettext-ed-string)
payments/payments/utils.py:429:14: W9904: gettext-ed string with positional args 'Address verification failed for postcode {}' (bad-gettext-ed-string)
************* Module payments.payments.hooks.processors
payments/payments/hooks/processors.py:116:-1: W9904: gettext-ed string with positional args '\nKYC submission failed\n\nDashboard verification:\n{}\n\nMangopay dashboard:\n{}\n            ' (bad-gettext-ed-string)
payments/payments/hooks/processors.py:152:18: W9904: gettext-ed string with positional args 'Transaction not found for Payin {}' (bad-gettext-ed-string)
payments/payments/hooks/processors.py:158:18: W9904: gettext-ed string with positional args 'The transaction {} (payin:{}) was paid in a different currency {}' (bad-gettext-ed-string)
payments/payments/hooks/processors.py:181:18: W9904: gettext-ed string with positional args 'The transaction {} (payin:{}) was expected to wire {} but instead wired {}' (bad-gettext-ed-string)
payments/payments/hooks/processors.py:298:-1: W9904: gettext-ed string with positional args '\nMangopay transaction details:\n{}\n                    ' (bad-gettext-ed-string)
************* Module payments.payments.api.v2.property_mixins
payments/payments/api/v2/property_mixins.py:52:18: W9904: gettext-ed string with positional args 'Organisation {} has no registered wallet' (bad-gettext-ed-string)
************* Module payments.payments.api.v2.serializers
payments/payments/api/v2/serializers.py:138:18: W9904: gettext-ed string with positional args 'Transaction with uuid {} does not exists' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:250:18: W9904: gettext-ed string with positional args 'Pro {} has no registered wallet' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:385:18: W9904: gettext-ed string with positional args 'Error creating Card ({}) for user {}. Card already in the system.' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:430:18: W9904: gettext-ed string with positional args 'Error creating Card ({}) for user {}. Card already in the system.' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:570:22: W9904: gettext-ed string with positional args 'Wallet {} not assigned to any resource' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:659:22: W9904: gettext-ed string with positional args 'Wallet top up for org: {}' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:687:22: W9904: gettext-ed string with positional args 'Invalid Transaction state: {}' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:758:18: W9904: gettext-ed string with positional args 'Could not create CardRegistration. Error: {}' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:819:18: W9904: gettext-ed string with positional args 'Payment {} already {}' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:920:18: W9904: gettext-ed string with positional args 'Payment {} already {}' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:976:18: W9904: gettext-ed string with positional args 'Transaction with payment_request_uuid: {} not found in paid_in state' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:1013:18: W9904: gettext-ed string with positional args 'Pro {} has no registered wallet' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:1022:18: W9904: gettext-ed string with positional args "Pro's registered wallet ({}) not found" (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:1060:18: W9904: gettext-ed string with positional args 'Pro {} has no registered wallet' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:1098:18: W9904: gettext-ed string with positional args 'Could not withdraw funds: {}' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:1418:30: W9904: gettext-ed string with positional args 'Pro {}' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:1505:18: W9904: gettext-ed string with positional args 'Pro {} has no registered wallet' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:1592:30: W9904: gettext-ed string with positional args 'Customer {}' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:1656:30: W9904: gettext-ed string with positional args 'ExternalInvoice {}' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:1724:30: W9904: gettext-ed string with positional args 'Organisation {}' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:1793:18: W9904: gettext-ed string with positional args 'Organisation {} has no registered wallet' (bad-gettext-ed-string)
payments/payments/api/v2/serializers.py:1802:18: W9904: gettext-ed string with positional args "Organisation's registered wallet ({}) not found" (bad-gettext-ed-string)
************* Module payments.payments.api.v2.api_utils
payments/payments/api/v2/api_utils.py:25:14: W9904: gettext-ed string with positional args 'Invalid JSON request: {} empty or missing.' (bad-gettext-ed-string)
payments/payments/api/v2/api_utils.py:50:31: W9904: gettext-ed string with positional args 'Token {}' (bad-gettext-ed-string)
payments/payments/api/v2/api_utils.py:88:31: W9904: gettext-ed string with positional args 'Token {}' (bad-gettext-ed-string)
```

--- a/missing_gettext.py
+++ b/missing_gettext.py
@@ -40,6 +40,9 @@ import string
 import os.path
 
 
+bad_format_regex = re.compile('{[0-9]*}')
+
+
 def is_number(text):
     """Returns True if this text is a representation of a number"""
     try:
@@ -424,7 +427,7 @@ class MissingGettextChecker(BaseChecker):
                             ]):
                             # we're in a _() call
                             string_ok = True
-                            if '{}' in node.value:
+                            if bad_format_regex.search(node.value):
                                 self.add_message('W9904', node=node, args=node.value)
                             break
 

